src/LaunchDarkly.ClientSdk/LdClient.cs

```diff
@@ -317,7 +358,53 @@ async Task StartAsync()
         /// <remarks>
         /// <para>
         /// The returned task will yield the <see cref="LdClient"/> instance once the first response from
-        /// the LaunchDarkly service is returned (or immediately if it is in offline mode).
+        /// the LaunchDarkly service is returned or immediately if it is offline.
+        /// </para>
+        /// <para>
+        /// To specify additional configuration options rather than just the mobile key, you can use
+        /// <see cref="Init(Configuration, Sdk.Context, TimeSpan)"/> or <see cref="InitAsync(Configuration, Sdk.Context)"/>.
+        /// </para>
+        /// <para>
+        /// If you would rather a synchronous version of this method, use
+        /// <see cref="Init(string, ConfigurationBuilder.AutoEnvAttributes, Sdk.Context, TimeSpan)"/>.
+        /// </para>
+        /// <para>
+        /// You must use one of these static factory methods to instantiate the single instance of LdClient
+        /// for the lifetime of your application.
+        /// </para>
+        /// </remarks>
+        /// <param name="mobileKey">the mobile key given to you by LaunchDarkly</param>
+        /// <param name="autoEnvAttributes">Enable / disable Auto Environment Attributes functionality.  When enabled,
+        /// the SDK will automatically provide data about the environment where the application is running.
+        /// This data makes it simpler to target your mobile customers based on application name or version, or on
+        /// device characteristics including manufacturer, model, operating system, locale, and so on. We recommend
+        /// enabling this when you configure the SDK.  See
+        /// <a href="https://docs.launchdarkly.com/sdk/features/environment-attributes">our documentation</a> for
+        /// more details.</param>
+        /// <param name="initialContext">the initial evaluation context; see <see cref="LdClient"/> for more
+        /// about setting the context and optionally requesting a unique key for it</param>
+        /// <returns>a Task that resolves to the singleton LdClient instance</returns>
+        [Obsolete("Initializing the LDClient without a timeout is no longer permitted to help prevent" +
+                  "consumers from blocking their application execution by mistake when connectivity is poor.  Please" +
+                  "use InitAsync(string, ConfigurationBuilder.AutoEnvAttributes, Context, TimeSpan) and specify a max wait time.")]
```

Comment:
For reviewers: The diff engine makes this look like I added a method and deprecated it immediately.  That is not the case, just the diff shifting things around.